src/components/App.jsx

```diff
@@ -33,15 +52,53 @@ const App = () => {
         }
     };
 
+    const updateGroceryListItem = async (item) => {
+        try {
+            const resp = await fetch(`http://localhost:4000/shopItem`, {
+                method: "PUT",
+                mode: "cors",
+                headers: {
+                    "Content-type": "application/json"
+                },
+                body: JSON.stringify(item)
+            })
+            if (resp.ok) {
+                fetchGroceryList("isCompleted=false");
+            }
+        }
+        catch (error) {
+            console.log(error)
+        }
+    }
+
+    const deleteItemFromGroceryList = async (id) => {
+        try {
+            const resp = await fetch(`http://localhost:4000/shopItem/${id}`, {
+                method: "DELETE",
+                mode: "cors",
+                headers: {
+                    "Content-type": "application/json"
+                },
+            })
+            if (resp.ok) {
+                fetchGroceryList("isCompleted=false")
+            }
+        }
+        catch (error) {
+            console.log(error)
+        }
+    }
+
     return (
         <div className='flex flex-col col-start-2 bg-slate-50 shadow-xl gap-4 overflow-y-scroll no-scrollbar'>
             <Head database={groceryList}></Head>
             <ListButtons fetch={fetchGroceryList}></ListButtons>
             <ListItems
                 items={groceryList}
-                onEdited={(thisItem) => console.log(thisItem)}
+                onEdited={(item) => updateGroceryListItem(item)}
+                onDeleted={(id) => deleteItemFromGroceryList(id)}
             ></ListItems>
-            <NewButton></NewButton>
+            <NewButton fetch={(item) => sendItem(item)}></NewButton>
```

Comment:
I'd rename fetch to onAdd or something like that... It describe the event better.

Also item to name. Alternatively you could construct `{ name: prompt, isCompleted: false }` in NewButton as well instead of line 25 in here which would be more consistent with the rest of the App (e.g. ListItem also sends the whole object back).